src/ansys/dpf/gate/load_api.py

```diff
@@ -51,38 +51,51 @@ def _pythonize_awp_version(version):
     return "20" + version[0:2] + "." + version[2]
 
 
-def _find_latest_ansys_versions():
-    awp_versions = [key[-3:] for key in os.environ.keys() if "AWP_ROOT" in key]
-    installed_packages_list = {}
+def _find_latest_dpf_server():
+    path_per_version = {}
 
-    for awp_version in awp_versions:
-        if not awp_version.isnumeric():
-            continue
-        ansys_path = os.environ.get("AWP_ROOT" + awp_version)
-        if ansys_path:
-            installed_packages_list[
-                packaging.version.parse(_pythonize_awp_version(awp_version))
-            ] = ansys_path
+    path_per_version = _paths_to_dpf_in_unified_installs(path_per_version)
+
+    path_per_version = _paths_to_dpf_server_library_installs(path_per_version)
+
+    if len(path_per_version) > 0:
+        return path_per_version[sorted(path_per_version)[-1]]
 
+
+def _paths_to_dpf_server_library_installs(path_per_version: dict) -> dict:
     installed_packages = pkg_resources.working_set
     for i in installed_packages:
         if "ansys-dpf-server" in i.key:
             file_name = pkg_resources.to_filename(i.project_name.replace("ansys-dpf-", ""))
             try:
                 module = importlib.import_module("ansys.dpf." + file_name)
-                installed_packages_list[
+                path_per_version[
                     packaging.version.parse(module.__version__)
                 ] = module.__path__[0]
             except ModuleNotFoundError:
                 pass
             except AttributeError:
                 pass
-    if len(installed_packages_list) > 0:
-        return installed_packages_list[sorted(installed_packages_list)[-1]]
+    return path_per_version
 
 
-def _unified_installer_path_if_exists():
-    return _find_latest_ansys_versions()
+def _paths_to_dpf_in_unified_installs(path_per_version: dict) -> dict:
+    awp_versions = [key[-3:] for key in os.environ.keys() if "AWP_ROOT" in key]
+    for awp_version in awp_versions:
+        if not awp_version.isnumeric():
+            continue
+        ansys_path = os.environ.get("AWP_ROOT" + awp_version)
+        if ansys_path:
+            # Check that this ansys path exists
+            if not os.path.isdir(ansys_path):
+                continue
+            # Check that it contains a DPF install
+            if not os.path.exists(os.path.join(ansys_path, _get_path_in_install())):
```

Comment:
`_get_path_in_install` should be enriched to also check the existence of the `dpf` folder, where the majority of libraries are now placed

Reply:
@rafacanton we could, yet checks are usually made in methods which call `_get_path_in_install`:
- `gate.load_api._get_api_path_from_installer_or_package`, 
- `core.server_types._run_launch_server_process`, 
- `core.server_types._verify_ansys_path_is_valid`, 
- `core.server_types._get_dll_path`, 
- `core.server_factory.ServerFactory.get_server_type_from_config`,

That would mean refactoring all of those.
Still I agree that the current checks may not be enough. We should check that a given list of DLLs (TBD) is actually present.
This can be added to the objectives of this issue/PR, what do you think?

Reply:
@rafacanton while implementing a test for `dpf/bin/winx64` or `dpf/bin/linx64` folder, I realized that we may have an issue in `ServerFactory.get_server_type_from_config` which basically enforces use of `AWP_ROOT242` when in non-legacy Grpc to define the aisol path to add to PATH... I think this is quite a problem. I also noticed that we still use `_find_outdated_ansys_version()` which tests for Ansys versions below 221... I do not think we still need that as it was used to force `LegacyGrpc` connections for servers 221 or below while now we could argue that the client itself does not support servers below 222. I think there may be a lot of refactoring to do here.

Reply:
Note: adding the test for a dpf folder, yet **it appears consistently only starting with 231**.